src/app/fdctl/run/tiles/fd_gossip.c

```diff
@@ -0,0 +1,724 @@
+  ctx->gossip_config.private_key = ctx->identity_private_key;
+  ctx->gossip_config.public_key = &ctx->identity_public_key;
+  ctx->gossip_config.deliver_fun = gossip_deliver_fun;
+  ctx->gossip_config.deliver_arg = ctx;
+  ctx->gossip_config.send_fun = gossip_send_packet;
+  ctx->gossip_config.send_arg = ctx;
+  ctx->gossip_config.sign_fun = gossip_signer;
+  ctx->gossip_config.sign_arg = ctx;
+  ctx->gossip_config.shred_version = 0;
+
+  if( fd_gossip_set_config( ctx->gossip, &ctx->gossip_config ) ) {
+    FD_LOG_ERR( ( "error setting gossip config" ) );
+  }
+
+  fd_gossip_peer_addr_t gossip_peer_addr;
+  FD_LOG_NOTICE(( "gossip initial peer - addr: %s", ctx->gossip_peer_addr ));
+  if( fd_gossip_add_active_peer( ctx->gossip, resolve_hostport( ctx->gossip_peer_addr, &gossip_peer_addr ) ) ) {
+    FD_LOG_ERR( ( "error adding gossip active peer" ) );
+  }
+
+  fd_gossip_update_addr( ctx->gossip, &ctx->gossip_config.my_addr );
+
+  fd_gossip_peer_addr_t tvu_my_addr;
+  fd_gossip_peer_addr_t tvu_my_fwd_addr;
+  fd_gossip_peer_addr_t tpu_my_addr;
+  fd_gossip_peer_addr_t tpu_vote_my_addr;
+  if( resolve_hostport( tile->gossip.tvu_my_addr, &tvu_my_addr ) == NULL ) {
+    FD_LOG_ERR( ( "error parsing tvu addr" ) );
+  }
+
+  if( resolve_hostport( tile->gossip.tvu_my_fwd_addr, &tvu_my_fwd_addr ) == NULL ) {
+    FD_LOG_ERR(( "error parsing tvu fwd addr" ) );
+  }
+
+  if( resolve_hostport( tile->gossip.tpu_udp_my_addr, &tpu_my_addr ) == NULL ) {
+    FD_LOG_ERR(( "error parsing tpu addr" ) );
+  }
+
+  if( resolve_hostport( tile->gossip.tpu_vote_udp_my_addr, &tpu_vote_my_addr ) == NULL ) {
+    FD_LOG_ERR(( "error parsing tpu vote addr" ) );
+  }
+
+  fd_gossip_update_tvu_addr( ctx->gossip, &tvu_my_addr, &tvu_my_fwd_addr );
+  fd_gossip_update_tpu_addr( ctx->gossip, &tpu_my_addr );
+  fd_gossip_update_tpu_vote_addr( ctx->gossip, &tpu_vote_my_addr );
+  fd_gossip_settime( ctx->gossip, fd_log_wallclock() );
+  fd_gossip_start( ctx->gossip );
+
+  FD_LOG_NOTICE(( "gossip listening on port %u", tile->gossip.gossip_listen_port ));
+}
+
+static ulong
+populate_allowed_seccomp( void *               scratch,
+                          ulong                out_cnt,
+                          struct sock_filter * out ) {
+  (void)scratch;
+  populate_sock_filter_policy_gossip( out_cnt, out, (uint)fd_log_private_logfile_fd() );
+  return sock_filter_policy_gossip_instr_cnt;
+}
+
+static ulong
+populate_allowed_fds( void * scratch,
+                      ulong  out_fds_cnt,
+                      int *  out_fds ) {
+  (void)scratch;
+  if( FD_UNLIKELY( out_fds_cnt<2 ) ) FD_LOG_ERR(( "out_fds_cnt %lu", out_fds_cnt ));
+
+  ulong out_cnt = 0;
+  out_fds[ out_cnt++ ] = 2; /* stderr */
+  if( FD_LIKELY( -1!=fd_log_private_logfile_fd() ) )
+    out_fds[ out_cnt++ ] = fd_log_private_logfile_fd(); /* logfile */
+  return out_cnt;
+}
+
+fd_topo_run_tile_t fd_tile_gossip = {
+  .name                     = "gossip",
+  .mux_flags                = FD_MUX_FLAG_COPY,
```

Comment:
Does this need a manual publish flag? My understanding is that we don't need to publish every time